docs/_posts/2021-02-13-hudi-key-generators.md

```diff
@@ -5,18 +5,16 @@ author: shivnarayan
 category: blog
 ---
 
-Every record in Hudi is uniquely identified by a HoodieKey, which is a pair of record key and partition path where the 
-record belongs to. Hudi has imposed this constraint so that updates and deletes can be applied to the record of interest. 
-Hudi relies on the partition path field to partition your dataset and records within a partition have unique record keys. 
-Since uniqueness is guaranteed only within the partition, there could be records with same record keys across different 
-partitions. One should choose the partition field wisely as it could be a determining factor for your ingestion and 
-query latency.
+Every record in Hudi is uniquely identified by a primary key, which is a pair of record key and partition path where
```

Comment:
IMHO, we should mention HoodieKey only here rather than mentioning primary key. In the context of Hudi, HoodieKey is what uniquely identifies a particular record.

Reply:
We have plans to add column indexes in our roadmap and so we wanted to have this narrative where we use primary key for record keys or record key, partition path pairs and secondary keys as other columns. and hence.

Reply:
Yes makes sense in the backdrop of column/secondary keys